modules/core/exported/channel.go

```diff
@@ -34,6 +34,14 @@ type PacketI interface {
 // Acknowledgement defines the interface used to return
 // acknowledgements in the OnRecvPacket callback.
 type Acknowledgement interface {
+	// Success determines if the IBC application state should be persisted when handling `RecvPacket`.
+	// During `OnRecvPacket` IBC application callback execution, all state changes are held in a cache store and committed later.
+	//
+	// Note 1: IBC application callback events are always persisted so long as `RecvPacket` succeeds without error.
+	//
+	// Note 2: This method is independent of application level success/error which is encoded in the acknowledgement bytes in a protocol specific way.
```

Comment:
I'm not sure what this note means? Could you elaborate?

Reply:
You have a public type called "Acknowledgement" which is not an IBC acknowledgement. And `acknowledgement.Success` has nothing to do with success acknowledgements. This is a source of confusion for users that I tried to resolve.

Reply:
Ah, by "This method is independent of application level success/error" this is referencing [result](https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/types/acknowledgement.go#L22) and error acknowledgements, not the success/error of a transaction at the application layer?

> And acknowledgement.Success has nothing to do with success acknowledgements

Hmm, I'm not sure I agree with this point. If by "success acknowledgement" you mean "result acknowledgement" then the only reason it is referred to "success" is because it returns true when fulfilling the `acknowledgement.Success()` interface function. The response/error acknowledgements are not required, they are sane default acknowledgements IBC apps may use if they wish not to define their own custom types

After reading this again, are you referring to [this](https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/types/ack.go#L12)?

Reply:
> by "This method is independent of application level success/error" this is referencing [result](https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/types/acknowledgement.go#L22) and error acknowledgements, not the success/error of a transaction at the application layer?

Right. I used "success" as the opposite to an error acknowledgement.

I think my primary source of confusion is that the "Acknowledgement" interface **is not** the acknowledgement but **contains** the acknowledgement.

Reply:
Okay I see now. Thanks for the followup comments. I see how this can be confusing, especially given that an acknowledgement can wrap other acknowledgements. The idea of the interface is to convey information about the acknowledgements so that core IBC can do some useful logic without having reference to the actual type. That's why we need the `Success()` function and the `Acknowledgement()` function. So that core IBC can do some useful handling of state during receive packets and also set the marshalled bytes

Maybe we could make some more adjustments to clear up the confusion (I also see we need to add docs about wrapping acknowledgements). I will add some suggestions